### PR TITLE
[runtime env] use pytest tmp_path, os.path.sep, and unskip most tests for windows

### PR DESCRIPTION
**Pull request by mattip**
## Why are these changes needed?

Complements #22186 to enable runtime tests on windows. This refactors the tests to use the pytest `tmp_path` fixture, replaces '/' with `os.path.sep` and unskips almost all the tests on windows. One failing test is still skipped, I propose leaving this for a future PR.

## Checks

- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Review by architkulkarni — APPROVED** [review]
Looks great!

**Comment by architkulkarni** [comment]
Mac wheel build flaky on master, and //python/ray/tests:test_basic_2_client_mode flaky on master.

**Comment by mattip** [comment]
> Just to double check, does this running time look correct compared to running locally? 

Yes. Most of the tests are quite fast since they don't actually run `ray.init`

File: python/ray/tests/test_runtime_env_packaging.py [hunk 1]
```diff
@@ -247,103 +234,104 @@ def test_unzip_package(
...
-        dir_paths = set()
-        file_paths = set()
-        item_num = 0
-        excludes = []
-        root = Path(tmp_dir) / "test"
 
-        def construct(path, excluded=False, depth=0):
-            nonlocal item_num
-            path.mkdir(parents=True)
+@pytest.mark.skipif(sys.platform == "win32", reason="Fails on windows")
```
[the remark this answers is not in the document]
Comment (architkulkarni) [reply python/ray/tests/test_runtime_env_packaging.py:277]: I see, thanks for the details!  How odd...

[the remark this answers is not in the document]
Comment (architkulkarni) [reply python/ray/tests/test_runtime_env_packaging.py:277]: I think leaving this for a future PR makes sense.

**Pull request merged**